### PR TITLE
DROOLS-4847: [DMN Designer] Entries are expanded when user edit structures in a row

### PR DESCRIPTION
**Pull request by danielzhe**

### CODE REVIEW

**Comment by danielzhe** [comment]
I think this is a very good PR for @karreiro to review.
I had to make more changes in order to keep everything working.

**Comment by jomarko** [comment]
@danielzhe there is some conflict in merging.

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/types/listview/DataTypeListView.java [hunk 1]
```diff
@@ -326,6 +328,13 @@ public void insertAbove(final DataTypeListItem listItem,
 
         final Element elementReference = getDataTypeRow(reference);
         ElementHelper.insertBefore(listItem.getDragAndDropElement(), elementReference);
+        setNewElementYPosition(elementReference, listItem.getDragAndDropElement());
+    }
+
+    void setNewElementYPosition(final Element elementReference,
+                                final Element newElement) {
+        int referencePosition = getDndListComponent().getPositionY(elementReference);
```
Comment (jomarko) [inline kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/types/listview/DataTypeListView.java]: just for consistency, can we use `final int`?

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/types/listview/draganddrop/DNDListComponent.java [hunk 2]
```diff
@@ -138,6 +139,19 @@ public void refreshDragAreaSize() {
         view.refreshDragAreaSize();
     }
 
+    public void setInitialPositionY(final HTMLElement dragAndDropElement,
+                                    final List<HTMLElement> children) {
+        Integer parentY = Position.getY(dragAndDropElement);
+        for (final Element element : children) {
+            parentY++;
+            Position.setY(element, parentY);
+        }
```
Comment (jomarko) [inline kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/types/listview/draganddrop/DNDListComponent.java]: This is the same as below, but I am fine if you want to keep as it is.
```java
for (final Element element : children) {
            Position.setY(element, ++parentY);
        }
```

**Pull request merged**